### PR TITLE
evm: revert on selfdestruct opcode

### PR DESCRIPTION
**Pull request by ptrus**

### CODE REVIEW

**Comment by Yawning** [comment]
I get that a revert is a revert, but should this ensure that nothing screwed up is happening to the contract's storage?

**Review by kostko — COMMENTED** [review]
Also update the README in `modules/evm/README.md` to note that SELFDESTRUCT will cause a revert.

**Comment by kostko** [comment]
> I get that a revert is a revert, but should this ensure that nothing screwed up is happening to the contract's storage?

In case of an error being returned from a transaction, our handler will cause any storage changes to be rolled back. So it will be as if the transaction never happened (except the fee payments).

File: tests/benchmark/run-benchmarks.sh [hunk 1]
```diff
@@ -60,4 +60,4 @@ sleep 2
     --address "${CLIENT_SOCK}" \
     --runtime.id "8000000000000000000000000000000000000000000000000000000000000000" \
     --benchmarks accounts_transfers \
-    --benchmarks.concurrency 4000
+    --benchmarks.concurrency 3000
```
Comment (ptrus) [inline tests/benchmark/run-benchmarks.sh:63]: Not related to changes here, but hopefully this reduces the flakiness of the benchmarks CI job (works fine locally).

**Review by kostko — APPROVED** [review]

**Pull request merged**